modules/fedimint-ln-server/src/lib.rs

```diff
@@ -542,8 +542,24 @@ impl ServerModule for Lightning {
                         amount: amount.amount,
                         contract: contract.contract.clone().to_funded(out_point),
                     });
-                dbtx.insert_entry(&contract_db_key, &updated_contract_account)
+                let previous = dbtx
+                    .insert_entry(&contract_db_key, &updated_contract_account)
                     .await;
+                let inserted = previous.is_none();
+                if inserted {
```

Comment:
I assume if this value is overwritten (i.e previous is not none) that's unexpected behavior? Should we explicitly check for that?